### PR TITLE
 information_schema: Fill information_schema.key_column_usage

### PR DESCRIPTION
**Pull request by shenli**
Add content for key-column-usage table.
@zimulala @tiancaiamao @XuHuaiyu PTAL

### CODE REVIEW

File: infoschema/tables.go [hunk 1]
```diff
@@ -847,6 +848,103 @@ func dataForTableConstraints(schemas []*model.DBInfo) [][]types.Datum {
 	return rows
 }
 
+func dataForKeyColumnUsage(schemas []*model.DBInfo) [][]types.Datum {
+	rows := [][]types.Datum{}
+	for _, schema := range schemas {
+		for _, table := range schema.Tables {
+			rs := keyColumnUsageInTable(schema, table)
+			for _, r := range rs {
+				rows = append(rows, r)
```
Comment (hanfei1991) [inline infoschema/tables.go]: `rows = append(rows, rs...)`

File: infoschema/tables.go [hunk 2]
```diff
@@ -847,6 +848,103 @@ func dataForTableConstraints(schemas []*model.DBInfo) [][]types.Datum {
 	return rows
 }
 
+func dataForKeyColumnUsage(schemas []*model.DBInfo) [][]types.Datum {
+	rows := [][]types.Datum{}
```
Comment (hanfei1991) [inline infoschema/tables.go]: init the cap

File: infoschema/tables.go [hunk 3]
```diff
@@ -847,6 +848,103 @@ func dataForTableConstraints(schemas []*model.DBInfo) [][]types.Datum {
...
+			col := nameToCol[key.Name.L]
+			record := types.MakeDatums(
+				catalogVal,    // CONSTRAINT_CATALOG
+				schema.Name.O, // CONSTRAINT_SCHEMA
+				idxName,       // CONSTRAINT_NAME
+				catalogVal,    // TABLE_CATALOG
+				schema.Name.O, // TABLE_SCHEMA
+				table.Name.O,  // TABLE_NAME
+				col.Name.O,    // COLUMN_NAME
+				i+1,           // ORDINAL_POSITION,
```
Comment (tiancaiamao) [inline infoschema/tables.go]: why it's `i+1` here, but line 877 is `i` ?

Comment (shenli) [inline infoschema/tables.go]: Line 877 should be 1.
The value of ORDINAL_POSITION is the column's position within the constraint, not the column's position within the table. Column positions are numbered beginning with 1.
So for PkAsHandle, it should always be 1.

**Comment by shenli** [comment]
@tiancaiamao @hanfei1991 PTAL

**Comment by tiancaiamao** [comment]
LGTM

File: infoschema/tables.go [hunk 4]
```diff
@@ -847,6 +848,101 @@ func dataForTableConstraints(schemas []*model.DBInfo) [][]types.Datum {
 	return rows
 }
 
+func dataForKeyColumnUsage(schemas []*model.DBInfo) [][]types.Datum {
+	rows := make([][]types.Datum, len(schemas)) // The capacity is not accurate, but it is not a big problem.
```
Comment (zimulala) [inline infoschema/tables.go]: -> `make([][]types.Datum, 0, len(schemas))` ?

**Comment by zimulala** [comment]
LGTM

**Pull request merged**